hardware_interface/src/resource_manager.cpp

```diff
@@ -310,6 +310,54 @@ std::unordered_map<std::string, status> ResourceManager::get_components_status()
   return resource_storage_->hardware_status_map_;
 }
 
+void ResourceManager::prepare_command_mode_switch(
+  const std::vector<std::string> & start_interfaces,
+  const std::vector<std::string> & stop_interfaces)
+{
+  for (auto & component : resource_storage_->actuators_) {
+    if (return_type::OK !=
+      component.prepare_command_mode_switch(start_interfaces, stop_interfaces))
+    {
+      throw std::runtime_error(
```

Comment:
I am not certain if we should throw an exception here. This will break `ros2_control_node`. Maybe better to output error and stop hardware interface? What do you think?

In the future, we will potentially manage multiple hardware interfaces, so if the node breaks, it breaks for all of them.

Reply:
I'll switch to returning a `bool` to the `controller_manager` and outputting the error information on `RCLCPP_ERROR(rclcpp::get_logger("controller_manager"), msg)`. This ties the resource manager a little closer to the specific controller manager implementation, and I think exceptions have been thrown previously to pass error information between the resource manager and the controller manager without needing to knowing which logger it should pass the information to.